squiRL/common/data_stream.py

```diff
@@ -84,43 +80,53 @@ class RLDataset(IterableDataset):
 
     Attributes:
         agent (Agent): Agent that interacts with env
-        env (gym.Env): OpenAI gym environment
+        episodes_per_batch (int): number of episodes per batch
         net (nn.Module): Policy network
+        num_envs (int): Number of vectorized envs
         replay_buffer: Replay buffer
     """
     def __init__(self, replay_buffer: RolloutCollector,
-                 episodes_per_batch: int, net: MLP, agent) -> None:
+                 episodes_per_batch: int, net: MLP, agent,
+                 num_envs: int) -> None:
         """Summary
 
         Args:
             replay_buffer (RolloutCollector): Description
-            env (gym.Env): OpenAI gym environment
+            episodes_per_batch (int): number of episodes per batch
             net (nn.Module): Policy network
+            num_envs (int): Number of vectorized envs
             agent (Agent): Agent that interacts with env
         """
         self.replay_buffer = replay_buffer
         self.episodes_per_batch = episodes_per_batch
+        # self.steps_per_batch = episodes_per_batch
         self.net = net
         self.agent = agent
+        self.num_envs = num_envs
 
     def populate(self) -> None:
         """
         Samples an entire episode
 
         """
-        done = False
-        while not done:
-            reward, done = self.agent.play_step(self.net)
+        self.total_episodes_sampled = -np.ones([self.num_envs])
```

Comment:
if you have episodes_per_batch = 4 for example, then this loop would go 8 times because you initialise the count at -1. Is this intended?

Example:  
-1 -1 -1 -1 and after a True True False False  
0 0 -1 -1  

This results in a sum of -2. After 6 more True returns, this would be equal to 4. A total of 8 True firsts needed for the loop to end.

Reply:
This is on purpose because an episode is between 2 Trues. So we have 4 episodes between 8 Trues. The reason I do this is because I am not resetting all envs before I start sampling. So I want to ensure that I am sampling episodes that were entirely done under the same policy, not partial episodes.

Let me know if you think there's a better way of doing this. I just saw your commit for `reset_all()`. I'll try it now. If the overhead is significant, I won't make it the default use. Otherwise, I will and this would save us the hassle of dealing with partial episodes.

Reply:
I change this. Now Each episode has one True at the end. So the count starts from 0 not -1.